### DISCUSSION TITLE
loading addon-console takes a very long time

### DISCUSSION THREAD

**Discussion by unional** [storybookjs/builder-vite#524] [Q&A] [answered]
Hi,
I have setup my storybook with addons like `addon-console` and `addon-storysource`.

When running `start-storybook`, the page shows up but it takes a very long time before it renders the result.

```sh
╭────────────────────────────────────────────────────╮
│                                                    │
│   Storybook 6.5.13 for React started               │
│   3.9 s for preview                                │
│                                                    │
│    Local:            http://localhost:6200/        │
│    On your network:  http://192.168.86.58:6200/    │
│                                                    │
╰────────────────────────────────────────────────────╯
2:48:18 PM [vite] ✨ new dependencies optimized: @storybook/addon-console
2:48:18 PM [vite] ✨ optimized dependencies changed. reloading
^Cpnpm sb  18.69s user 9.95s system 126% cpu 22.611 total
```

The optimized step is skipped if restarting storybook the next time.

But it is quite often that it is invalidated so the optimization needs to be redone.

Do not observe the same slowness in webpack5.

**Comment by IanVS** [comment, marked as answer]
You can add `@storybook/addon-console` to `optimizeDeps.include` in a `viteFinal` in your `.storybook/main.cjs` if you'd like, that will help improve the time to start up.  But vite is just slower to start up than webpack in general, as well: https://storybook.js.org/blog/storybook-performance-from-webpack-to-vite/

**Reply by unional** [reply]
Thanks! Will try out the `optimizeDeps.include`:

```js
async viteFinal(config) {
  return mergeConfig(config, {
    optimizeDeps: {
      include: ['@storybook/addon-console']
    },
   // other configs
  }
}
```
